golos.publication/golos.publication.cpp

```diff
@@ -557,13 +570,15 @@ void publication::check_upvote_time(uint64_t cur_time, uint64_t mssg_date) {
 }
 
 fixp_t publication::calc_available_rshares(name voter, int16_t weight, uint64_t cur_time, const structures::rewardpool& pool) {
-    elaf_t abs_w(elai_t(abs(weight)) / elai_t(config::_100percent));
-    tables::limit_table lims(_self, _self.value);
     auto token_code = pool.state.funds.symbol.code();
+    tables::limit_table lims(_self, _self.value);
     int64_t eff_vesting = golos::vesting::get_account_effective_vesting(config::vesting_name, voter, token_code).amount;
-    use_charge(lims, structures::limitparams::VOTE, token::get_issuer(config::token_name, token_code),
-        voter, eff_vesting, token_code, false, abs_w);
-    fixp_t abs_rshares = FP(eff_vesting) * abs_w;
+
+    auto abs_wc = use_charge(lims, structures::limitparams::VOTE, token::get_issuer(config::token_name,
+                               token_code), voter, eff_vesting, token_code, false, weight);
+
+    fixp_t abs_rshares = FP(eff_vesting) * elai_t(abs_wc);
```

Comment:
we need to multiply vesting to used power and divide to 100%: https://github.com/GolosChain/golos/blob/6e4f81824aa09794aa21c029dd6c67724d185020/libraries/chain/steem_evaluator.cpp#L1504-L1506